src/common/snippets/src/pass/common_optimizations.cpp

```diff
@@ -68,82 +82,87 @@ void CommonOptimizations::SplitDimensionM(const std::shared_ptr<ov::snippets::op
 
     const auto mm_shape = matmul0->get_shape();
     const auto m_dim = get_dim_M(mm_shape);  // M
-    const auto n_dim = mm_shape.back(); // N
+    const auto batch_dim =
+        std::accumulate(mm_shape.rbegin() + 2, mm_shape.rend(), size_t(1), std::multiplies<size_t>());  // B (batch)
     // [113745] Heurestic is equal to double block size.
     // When this optimization will be moved into Subgraph and blocking param will be implemented as dependents of shapes,
     // need to implement common way (for all backends) to calculate optimal value of M dimension
     const auto optimal_m_dim = 32 * 2;
-    const auto optimal_parallelism_work_amount = minimal_concurrency;
-    if (m_dim <= optimal_m_dim)
+    const auto optimal_parallelism_work_amount = concurrency;
+    // We skip optimization
+    //  - if the current M dimension is not enough for splitting
+    //  - if the current batch is optimal for concurrency
+    if (m_dim <= optimal_m_dim || batch_dim % optimal_parallelism_work_amount == 0)
```

Comment:
It seems really strange that `optimal_m_dim` is hard-coded and doesn't depend on the number of threads or the actual selected block size. For example, if the shape is `[4, 60, 512]` and there are 16 threads available, it would still make sense to reshape into `[16, 15, 512]` to load all the threads. 
Moreover, is the `M` dim is too small, it's better to disable M-blocking (we don't need it in this case), and load all the threads instead.
What do you think?

Reply:
We decided to leave it so in the [previous](https://github.com/openvinotoolkit/openvino/pull/18160#discussion_r1261054283) review. Your comment makes a sense, right. Thanks a lot for the idea!
As you know we implemented this optimization for special model with specific shapes. If we get new model or new shapes, we can adjust the condition. Unfortunately we don't have enough time to additionally validate this idea before release. At the moment we have a good performance and I think it's enough still. Moreover, this code was before. I tried to just update second step to not break working optimization of first step in bounds of the current PR. Thank you!

Reply:
Discussed offline. Agreed to allow for splitting the only M block (remove `optimal_m_dim`). Need to perform validation.

Reply:
As we discussed offline, I removed `optimal_m_dim` and validated on model set. The results are attached to the ticket